src/loader.ts

```diff
@@ -81,8 +81,21 @@ async function ESBuildLoader(
 		} else {
 			/* Detect tsconfig file */
 
-			// Webpack shouldn't be loading the same path multiple times so doesn't need to be cached
-			const tsconfig = getTsconfig(resourcePath, 'tsconfig.json', tsconfigCache);
+			let tsconfig;
+
+			try {
+				// Webpack shouldn't be loading the same path multiple times so doesn't need to be cached
+				tsconfig = getTsconfig(resourcePath, 'tsconfig.json', tsconfigCache);
+			} catch (error) {
+				if (resourcePath.split(path.sep).includes('node_modules')) {
+					this.emitWarning(
+						new Error(`[esbuild-loader] Error while discovering tsconfig.json in dependency: "${error?.toString()}"`),
+					);
```

Comment:
I tend to log more by default for cases such as these, but I'd defer to your preference as the maintainer of the project. Do you think there's value in logging this, or would it be preferable to silently ignore it?

For the original issue that sparked this PR, I don't think it would be a directly actionable warning but may provide a breadcrumb if someone is debugging an unusual output

Reply:
I think it's a good idea. I would appreciate it as a user.

For the original issue, I think the warning should direct the users to report bugs to the appropriate repo.

I'll update the error message in `get-tsconfig` to be more specific about which file it's trying to resolve.